packages/cipherstash-proxy-integration/src/walkthrough.rs

```diff
@@ -0,0 +1,949 @@
+/*
+ * Database Schema Diagram
+ *
+ *     ┌─────────────────────────┐
+ *     │    patients             │
+ *     │                         │
+ *     │ id (PK)  uuid           │
+ *     │ pii   encrypted(jsonb)  │
+ *     └─────────┬───────────────┘
+ *               │
+ *               │ 1:N
+ *               │
+ *    ┌──────────┴──────────┬─────────────────────┐
+ *    │                     │                     │
+ *    │ 1:N                 │ 1:N                 │
+ *    ▼                     ▼                     ▼
+ * ┌─────────────────────────────┐ ┌────────────────────────────┐
+ * │patient_medications          │ │patient_procedures          │
+ * │                             │ │                            │
+ * │ patient_id (FK)             │ │ patient_id (FK)            │
+ * │ medication encrypted(jsonb) │ │ procedure  encrypted(jsonb)│
+ * └─────────────────────────────┘ └────────────────────────────┘
+ *
+ * Reference Tables:
+ *
+ * ┌─────────────────┐    ┌─────────────────┐
+ * │  medications    │    │   procedures    │
+ * │                 │    │                 │
+ * │ id (PK)         │    │ id (PK)         │
+ * │ name            │    │ name            │
+ * │ description     │    │ description     │
+ * └─────────────────┘    │ code            │
+ *                        │ procedure_type  │
+ *                        └─────────────────┘
+ *
+ * Notes:
+ * - PII data in patients.pii is encrypted using EQL v2
+ * - PII data in patients.pii is encrypted using EQL v2
+ * - Junction tables store encrypted procedure/medication details
+ * - Foreign keys enforce referential integrity with CASCADE DELETE
+ * - Reference tables contain plaintext lookup data
+ */
+
+#[cfg(test)]
+mod tests {
+    use crate::common::{connect_with_tls, insert, query_by_params, reset_schema_to, table_exists, trace, PROXY};
+    use serde::Serialize;
+    use serde_json::Value;
+    use uuid::Uuid;
+
+    const SCHEMA: &'static str = r#"
```

Comment:
I'm in two minds about having the setup so different from the rest of the tests.

Reply:
I wonder if this is actually a new crate?
The schema and structure is very very good, it could be the basis for demos in other languages as well.